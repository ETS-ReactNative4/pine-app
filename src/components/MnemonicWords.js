import React, { Component } from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { BlurView } from 'react-native-blur';
import Icon from 'react-native-vector-icons/Ionicons';

import MnemonicWord from './MnemonicWord';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  list: {
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  blur: {
    position: 'absolute',
    top: -50,
    bottom: -50,
    left: -10,
    right: -10
  },
  revealWrapper: {
    position: 'absolute',
    alignItems: 'center'
  },
  revealView: {
    alignItems: 'center'
  },
  revealIcon: {
    color: '#007AFF',
    fontSize: 48
  },
  revealText: {
    color: '#007AFF'
  }
});

export default class MnemonicWords extends Component {
  _renderWords(words) {
    return words.map((word, index) => {
      return (
        <MnemonicWord key={index} index={index} word={word} />
      );
    });
  }

  _renderBlur() {
    return (
      <BlurView
        style={[styles.blur, this.props.blurStyle]}
        blurType='light'
        blurAmount={5}
      />
    );
  }

  _renderRevealButton() {
    return (
      <TouchableOpacity activeOpacity={0.7} onPress={this.props.onReveal} style={styles.revealWrapper}>
        <View style={styles.revealView}>
          <Icon name='ios-eye' style={styles.revealIcon} />
          <StyledText style={styles.revealText}>
            Reveal Recovery Key
          </StyledText>
        </View>
      </TouchableOpacity>
    );
  }

  render() {
    const phrase = this.props.phrase;
    const words = phrase.split(' ');

    return (
      <View style={[styles.list, this.props.style]}>
        {this._renderWords(words)}

        {this.props.revealed ? null : this._renderBlur()}
        {this.props.revealed ? null : this._renderRevealButton()}
      </View>
    );
  }
}

MnemonicWords.propTypes = {
  phrase: PropTypes.string.isRequired,
  style: PropTypes.any,
  blurStyle: PropTypes.any,
  onReveal: PropTypes.func,
  revealed: PropTypes.bool
};
