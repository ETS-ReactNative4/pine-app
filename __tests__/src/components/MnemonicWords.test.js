import React from 'react';
import renderer from 'react-test-renderer';
import MnemonicWords from '../../../src/components/MnemonicWords';

describe('MnemonicWords', () => {
  it('renders correctly', () => {
    const phrase = 'during bulb nominee acquire paddle next course stable govern eagle title wing';

    const tree = renderer.create(
      <MnemonicWords phrase={phrase} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
