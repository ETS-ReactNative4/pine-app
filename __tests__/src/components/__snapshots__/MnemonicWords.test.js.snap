// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnemonicWords renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        1
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        during
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        2
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        bulb
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        3
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        nominee
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        4
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        acquire
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        5
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        paddle
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        6
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        next
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        7
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        course
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        8
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        stable
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        9
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        govern
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        10
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        eagle
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        11
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        title
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#DADADA",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 11,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontSize": 11,
              "fontWeight": "bold",
              "left": 5,
              "position": "absolute",
              "textAlign": "right",
              "top": 11,
              "width": 20,
            },
          ]
        }
      >
        12
        .
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "color": "#999999",
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
              "paddingLeft": 17,
            },
          ]
        }
      >
        wing
      </Text>
    </View>
  </View>
</View>
`;
